### DISCUSSION TITLE
Optimalization: Importing values to theme or using babel-config-preval

### DISCUSSION THREAD

**Discussion by samuelhulla** [tailwindlabs/tailwindcss#8674] [Help]
Hey, I have a quick question regarding performance.

I need to reference a certain theme value inside a component outside of the static css, to provide a complex `framer-motion` animation for one of my components.

More specificly, I want to reference the default `theme` values for `screens` as in responsive breakpoints.

I know the official documentation says the following:

> To make this easy, Tailwind provides a resolveConfig helper you can use to generate a fully merged version of your configuration object:
>
> [*later...*]
>
> Note that this will transitively pull in a lot of our build-time dependencies, resulting in bigger bundle client-side size. To avoid this, we recommend using a tool like [babel-plugin-preval](https://github.com/kentcdodds/babel-plugin-preval) to generate a static version of your configuration at build-time.

Now I was thinking, given the nature of the `tailwind.config.js` file and the fact it is a simple module exporting `.js` file, wouldn't it be actually beneficial for performance to create a referential `.js/.ts` file that I would directly import into `tailwind.config.js` and use as constant reference in the config? 

And when needing to refernece the thee value, I could just import the individual values from the reference file isntead of importing and resolving the entire theme? 

___

Given the the skill and deeper understanding of the development team, I would presume it is better to write the values directly in the `tailwind.config.js` for reasons beyond my understanding and the usage with `babel-plugin-preval` is simply a must, but I'm just making sure, that there is no other better way, in which i would not have to import and resolve entire theme config file through helper functions.

**Comment by rohmann** [comment]
Hi @samuelhulla,

This might not be what you're looking for, and I hope this doesn't distract from you actual question. However, I'm recalling a few situations where I needed a component to pull from a dimension defined in a SASS variable, and I didn't want to redeclare that in my javascript. This can be accomplished for Tailwind theme values with a similar technique. First, you'd need to make sure the theme values are available as CSS custom properties like this:

```css
:root {
  --screens-sm: theme(screens.sm);
  --screens-md: theme(screens.md);
  --screens-lg: theme(screens.lg);
  --screens-xl: theme(screens.xl);
  --screens-2xl: theme(screens.2xl);
}
```

Then when your component renders, you can retrieve those values like this:

```js
const style = getComputedStyle(document.documentElement);
const screens = {
    sm: style.getPropertyValue('--screens-sm').trim(),
    md: style.getPropertyValue('--screens-md').trim(),
    lg: style.getPropertyValue('--screens-lg').trim(),
    xl: style.getPropertyValue('--screens-xl').trim(),
    '2xl': style.getPropertyValue('--screens-2xl').trim()
}
```

At that point `screens` will hold the pixel values coming from your tailwind config. If you've extended your config to include additional screen sizes, you'll need to include those as well. If you just need them as an array you could do this more succinctly:

```css
:root {
  --screens: theme(screens.sm) theme(screens.md) theme(screens.lg) theme(screens.xl) theme(screens.2xl);
}
```

```js
const screens = getComputedStyle(document.documentElement).getPropertyValue('--screens').split(' ').filter(size => !!size)
```
